scout/parse/config_base.py

```diff
@@ -0,0 +1,262 @@
+
+    # Each case has to have a owner. If not provided in config file it needs to be given as a
+    # argument
+    @validator("owner", pre=True, always=True)
+    def mandatory_check_owner(cls, value):
+        if value is None:
+            raise ConfigError("A case has to have a owner")
+        return value
+
+    @validator("family", pre=True, always=True)
+    def mandatory_check_family(cls, value):
+        if value is None:
+            raise ConfigError("A case has to have a 'family'")
+        return value
+
+    @validator("madeline_info")
+    def check_if_madeline_exists(cls, path):
+        """Add the pedigree figure, this is a xml file which is dumped in the db"""
+        mad_path = Path(path)
+        if not mad_path.exists():
+            raise ValueError("madeline path not found: {}".format(mad_path))
+        with mad_path.open("r") as in_handle:
+            return in_handle.read()
+
+    @validator("individuals")
+    def family_relations_consistent(cls, individuals):
+        individual_dicts = [i.dict() for i in individuals]
+        if len(individual_dicts) == 0:
+            raise PedigreeError("No samples could be found")
+        all_ids = [i["individual_id"] for i in individual_dicts]
+        # Check if relations are correct
+        for parsed_ind in individual_dicts:
+            father = parsed_ind.get("father")
+            if father and father != "0":
+                if father not in all_ids:
+                    raise PedigreeError("father %s does not exist in family" % father)
+            mother = parsed_ind.get("mother")
+            if mother and mother != "0":
+                if mother not in all_ids:
+                    raise PedigreeError("mother %s does not exist in family" % mother)
+        return individuals
+
+    @validator("track")
+    def field_not_none(cls, v):
+        if v is None:
+            raise ValueError("Owner and family can not be None")
+        return v
+
+    @validator("synopsis")
+    def synopsis_pre_hook(cls, my_synopsis):
+        if isinstance(my_synopsis, list):
+            return ". ".join(my_synopsis)
+        return my_synopsis
+
+    @root_validator
+    def update_track(cls, values):
+        """Performt hooks to config:
+        * Set track to 'cancer' if certain vcf-files are set.
+        * Make sure `collaborators` is set."""
+
+        # Handle special circumstances
+        vcfs = values.get("vcf_files")
+        try:
+            vcf_dict = vcfs.dict()
+            if (
+                vcf_dict["vcf_cancer"]
+                or vcf_dict["vcf_cancer_research"]
+                or vcf_dict["vcf_cancer_sv"]
+                or vcf_dict["vcf_cancer_sv_research"]
+            ):
+                values.update({"track": "cancer"})
+        except Exception as error:
+            LOG.error("exception in vcf_s! Not set? {}".format(error))
+
+        # Update collaborators to [owner] if not set
+        if values.get("collaborators") is None:
```

Comment:
Why is collaborator settings inside the VCF track validator? These fields are not related